pages/_app.js

```diff
@@ -0,0 +1,62 @@
+import App, { Container } from 'next/app';
+import React from 'react';
+import { Provider } from 'react-redux';
+import withRedux from 'next-redux-wrapper';
+import withReduxSaga from 'next-redux-saga';
+import { SAVE_SESSION_REQUEST, FETCH_USER_REQUEST } from '../components/auth/actionTypes';
+import createStore from "../store";
+import Layout from '../components/layout';
+
+class MyApp extends App {
+    static async getInitialProps({ Component, ctx }){
+        let pageProps = {};
+        const { req, store, isServer } = ctx;
+        
+        //  if in server, get cookie and fetch user
+        if (isServer){
+            console.log(req.headers.cookie);
+            const { cookie } = req.headers;
+            if (cookie){
+                await store.dispatch({ payload: { cookie }, type: SAVE_SESSION_REQUEST });
+                await store.dispatch({ type: FETCH_USER_REQUEST });
```

Comment:
define actions in an actions.js file

Reply:
do not await these actions here

Reply:
I have to use await. I had issues when I didn't use it before.. spent hours debugging. WIthout it, the user data isn't fetched before the admin page is rendered and that results in an error.

Reply:
Okay i get you but this pattern is incorrect. using `await` signifies as asynchronous operation. and in the redux pattern sagas handle all async operations. So what you need to do here is trigger an action that is consumed by a saga then do whatever other dispatching that you need to do there. DO NOT do it in the component it self